components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -4115,6 +4115,48 @@ public int compare(Application o1, Application o2) {
         return applications;
     }
 
+    public Application[] getAllApplicationsOfTenantForMigration(String appTenantDomain) throws
+            APIManagementException {
+
+        Connection connection;
+        PreparedStatement prepStmt = null;
+        ResultSet rs;
+        Application[] applications = null;
+        String sqlQuery = SQLConstants.GET_SIMPLE_APPLICATIONS;
+
+        String tenantFilter = "AND SUB.TENANT_ID=?";
+        sqlQuery += tenantFilter ;
+        try {
+            connection = APIMgtDBUtil.getConnection();
+
+            int appTenantId = APIUtil.getTenantIdFromTenantDomain(appTenantDomain);
+            prepStmt = connection.prepareStatement(sqlQuery);
+            prepStmt.setInt(1, appTenantId);
+            rs = prepStmt.executeQuery();
+
+            ArrayList<Application> applicationsList = new ArrayList<Application>();
+            Application application;
+            while (rs.next()) {
+                application = new Application(Integer.parseInt(rs.getString("APPLICATION_ID")));
+                application.setName(rs.getString("NAME"));
+                application.setOwner(rs.getString("CREATED_BY"));
+                applicationsList.add(application);
+            }
+            applications = applicationsList.toArray(new Application[applicationsList.size()]);
+        } catch (SQLException e) {
+            handleException("Error when reading the application information from" + " the persistence store.", e);
```

Comment:
can we remove this additional string concat

Reply:
Fixed with 1afea34